pylele_solid.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+"""
+    Abstract Base and Concrete Classes for all Gugulele Solid objects
+"""
+
+from __future__ import annotations
+
+import os
+import sys
+import argparse
+import datetime
+from pathlib import Path
+from abc import ABC, abstractmethod
+
+from pylele_api import ShapeAPI, Shape
+from pylele_config import Fidelity, Implementation, CARBON, GRAY, LITE_GRAY,DARK_GRAY, ORANGE, WHITE
+
+COLORS = {
```

Comment:
Can we make this an enum instead of just a map? I prefer it as enum offers stronger tool assisted checks, and ability to add additional methods and custom string representation.

Reply:
I guess we can... need to figure out exactly how, since the underlying datatype is a tuple.
I will investigate it.